### PR TITLE
[EX-4647] Mock adapter for partial results e2e tests

### PR DESCRIPTION
**Pull request by mavmaf**
# Pull Request Template #

This PR mocks the adapter for the partial results, changes the pertinent tests to make it use it and deletes from the view the components that were not longer needed. 

## Motivation and Context ##

Due to the variability of the real API that was being used for the e2e tests, the data was changing quite often causing the tests to fail. Mocking the adapter for the different components solves this issue, as we now control the data.

## What is the destination branch of this PR? ##

EX-4647

## How Has This Been Tested? ##

As usual, please change the following line in cypress.json and ensure everything works as expected:
"testFiles": ["cucumber/**/*.feature"],

### CODE REVIEW

**Review by CachedaCodes — COMMENTED** [review]
All good. Merge with main to solve conflicts and I'll approve

File: packages/x-components/tests/e2e/cucumber/no-suggestions/no-suggestions.feature [hunk 1]
```diff
@@ -1,16 +1,20 @@
 Feature: No Suggestions component
```
Comment (pmareke) [inline packages/x-components/tests/e2e/cucumber/no-suggestions/no-suggestions.feature:1]: Please remove this file.

File: packages/x-components/tests/e2e/cucumber/no-suggestions/no-suggestions.spec.ts [hunk 2]
```diff
@@ -1,7 +1,15 @@
 import { And, Given } from 'cypress-cucumber-preprocessor/steps';
```
Comment (pmareke) [inline packages/x-components/tests/e2e/cucumber/no-suggestions/no-suggestions.spec.ts:1]: Please remove this file.

File: packages/x-components/tests/e2e/cucumber/partial-results/partial-results.feature [hunk 3]
```diff
@@ -1,7 +1,11 @@
 Feature: Partial results component
 
+  Background:
```
Comment (pmareke) [inline packages/x-components/tests/e2e/cucumber/partial-results/partial-results.feature:3]: I think in this case we can remove the `Background` as we're doing more requests than we should and move this `Given` step to the Scenarios 2 and 3 just before the `And  no special config for partial-results view`.

File: packages/x-components/tests/e2e/cucumber/partial-results/partial-results.spec.ts [hunk 4]
```diff
@@ -1,20 +1,46 @@
...
 Given('no special config for partial-results view', () => {
-  cy.visit('test/partial-results');
+  cy.visit('test/partial-results?useMockedAdapter=true');
 });
 
 Then('at least {int} related results are displayed', (minResultsWithoutPartials: number) => {
   cy.getByDataTest('regular-result').should('have.length.at.least', minResultsWithoutPartials);
 });
 
 And('no partial results are displayed', () => {
```
Comment (pmareke) [inline packages/x-components/tests/e2e/cucumber/partial-results/partial-results.spec.ts:42]: `partial results are not displayed` ?

**Review by pmareke — CHANGES_REQUESTED** [review]

File: packages/x-components/tests/e2e/cucumber/partial-results/partial-results.spec.ts (hunk 4 above)
Comment (mavmaf) [reply packages/x-components/tests/e2e/cucumber/partial-results/partial-results.spec.ts:42]: I would leave it as it is per consistency with the rest of steps across the project (`no identifier results are displayed`, `no query suggestion are displayed`, `no history queries are displayed`...)

**Review by pmareke — APPROVED** [review]

**Review by CachedaCodes — APPROVED** [review]

**Pull request merged**